core/src/data/mbtilesDataSource.cpp

```diff
@@ -0,0 +1,422 @@
+            else if (name == "metadata") metadata = true;
+            else if (name == "geocoder_data") geocoder_data = true;
+            else if (name == "tiles") tiles = true;
+            else if (name == "grids") grids = true;
+            else if (name == "grid_data") grid_data = true;
+        }
+    } catch (std::exception& e) {
+        LOGE("Unable to check schema of SQLite MBTiles database: %s", e.what());
+    }
+
+    // Return if we have all the tables and views that should exist.
+    if (map && grid_key && keymap && grid_utfgrid && images &&
+            metadata && geocoder_data && tiles && grids && grid_data) {
+        return;
+    }
+
+    // Otherwise, we need to execute schema.sql to set up the db with the right schema.
+    try {
+        // Execute schema.
+        db.exec(SCHEMA);
+
+        // Fill in metadata table.
+        // https://github.com/pnorman/mbtiles-spec/blob/2.0/2.0/spec.md#content
+        // https://github.com/mapbox/mbtiles-spec/pull/46
+        SQLite::Statement stmt(db, "REPLACE INTO metadata (name, value) VALUES (?, ?);");
+
+        // name, type, version, description, format, compression
+        stmt.bind(1, "name");
+        stmt.bind(2, _name);
+        stmt.exec();
+        stmt.reset();
+
+        stmt.bind(1, "type");
+        stmt.bind(2, "baselayer");
+        stmt.exec();
+        stmt.reset();
+
+        stmt.bind(1, "version");
+        stmt.bind(2, 1);
+        stmt.exec();
+        stmt.reset();
+
+        stmt.bind(1, "description");
+        stmt.bind(2, "MBTiles tile container created by Tangram ES.");
+        stmt.exec();
+        stmt.reset();
+
+        stmt.bind(1, "format");
+        stmt.bind(2, _mimeType);
+        stmt.exec();
+        stmt.reset();
+
+        // Compression not yet implemented.
+        // http://www.iana.org/assignments/http-parameters/http-parameters.xhtml#content-coding
+        // identity means no compression
+        stmt.bind(1, "compression");
+        stmt.bind(2, "identity");
+        stmt.exec();
+
+    } catch (std::exception& e) {
+        LOGE("Unable to setup SQLite MBTiles database: %s", e.what());
+    }
+}
+
+bool MBTilesDataSource::getTileData(const TileID& _tileId, std::vector<char>& _data) {
+
+    auto& stmt = m_queries->getTileData;
+    try {
+        // Google TMS to WMTS
+        // https://github.com/mapbox/node-mbtiles/blob/
+        // 4bbfaf991969ce01c31b95184c4f6d5485f717c3/lib/mbtiles.js#L149
+        int z = _tileId.z;
+        int y = (1 << z) - 1 - _tileId.y;
+
+        stmt.bind(1, z);
+        stmt.bind(2, _tileId.x);
+        stmt.bind(3, y);
+
+        if (stmt.executeStep()) {
+            SQLite::Column column = stmt.getColumn(0);
+            const char* blob = (const char*) column.getBlob();
+            const int length = column.getBytes();
+
+            // OSM2Vectiles does not contain a 'compression' field
+            // Try deflate by default..
+            if (inflate(blob, length, _data) != 0) {
```

Comment:
Maybe when we open the database we should look and see if there is a compression field. If there is no compression field, I agree to try deflate by default and fall back to identity.